### PR TITLE
Move C fallback functions into arch/generic [Part 2]

### PR DESCRIPTION
**Pull request by Dead2**
This builds upon PR #1630, and needs to be rebased once that is in.

Moves these C functions into arch/generic
- Chunkset and related
- inflate_fast
- insert_string and update_hash
- slide_hash
- compare256 and longest_match

Also moves include files out of match_tpl.h to simplify dependency handling and avoid attempting to parse the includes more times than needed.

### CODE REVIEW

File: arch/generic/Makefile.in [hunk 1]
```diff
@@ -57,6 +58,13 @@ insert_string_c.o: $(SRCDIR)/insert_string_c.c  $(SRCTOP)/zbuild.h $(SRCTOP)/def
 insert_string_c.lo: $(SRCDIR)/insert_string_c.c  $(SRCTOP)/zbuild.h $(SRCTOP)/deflate.h $(SRCTOP)/insert_string_tpl.h
 	$(CC) $(SFLAGS) $(INCLUDES) -c -o $@ $(SRCDIR)/insert_string_c.c
 
+slide_hash_c.o: $(SRCDIR)/slide_hash_c.c  $(SRCTOP)/zbuild.h $(SRCTOP)/deflate.h
+	$(CC) $(CFLAGS) $(INCLUDES) -c -o $@ $(SRCDIR)/slide_hash_c.c
+
+slide_hash_c.lo: $(SRCDIR)/slide_hash_c.c  $(SRCTOP)/zbuild.h $(SRCTOP)/deflate.h
+	$(CC) $(SFLAGS) $(INCLUDES) -c -o $@ $(SRCDIR)/slide_hash_c.c
+
+
```
Comment (nmoinvaz) [inline arch/generic/Makefile.in:74]: Is two new lines here intentional?

**Review by nmoinvaz — APPROVED** [review]
LGTM

File: arch/generic/Makefile.in (hunk 1 above)
Comment (Dead2) [reply arch/generic/Makefile.in:74]: It was not intentionally added in that commit, but I had intended to keep 2 empty lines above and below that list from the beginning, to provide a bit more separation in the file. IMO it looks cleaner, but that is just my personal preference.

Comment (nmoinvaz) [reply arch/generic/Makefile.in:74]: Ok.

**Review by nmoinvaz — APPROVED** [review]

File: win32/Makefile.a64 [hunk 2]
```diff
@@ -168,26 +171,32 @@ gzwrite2.obj: gzwrite.c
 {$(TOP)/arch/arm}.c.obj:
 	$(CC) -c -I$(TOP) $(WFLAGS) $(CFLAGS) $<
 
+{$(TOP)/arch/generic}.c.obj:
+    $(CC) -c -I$(TOP) $(WFLAGS) $(CFLAGS) $<
```
Comment (mtl1979) [inline win32/Makefile.a64]: Indentation

File: win32/Makefile.arm [hunk 3]
```diff
@@ -189,26 +192,32 @@ gzwrite2.obj: gzwrite.c
 {$(TOP)/arch/arm}.c.obj:
 	$(CC) -c -I$(TOP) $(WFLAGS) $(CFLAGS) $<
 
+{$(TOP)/arch/generic}.c.obj:
+    $(CC) -c -I$(TOP) $(WFLAGS) $(CFLAGS) $<
```
Comment (mtl1979) [inline win32/Makefile.arm]: Indentation

File: win32/Makefile.a64 (hunk 2 above)
Comment (Dead2) [reply win32/Makefile.a64]: Fixed in #1630

File: win32/Makefile.arm (hunk 3 above)
Comment (Dead2) [reply win32/Makefile.arm]: Fixed in #1630

**Pull request merged**